pinot-plugins/pinot-input-format/pinot-clp-log/src/main/java/org/apache/pinot/plugin/inputformat/clplog/CLPLogMessageDecoder.java

```diff
@@ -38,18 +38,27 @@
  * details. The implementation is based on {@link org.apache.pinot.plugin.inputformat.json.JSONMessageDecoder}.
  */
 public class CLPLogMessageDecoder implements StreamMessageDecoder<byte[]> {
+  public static final String ERROR_SAMPLING_PERIOD_CONFIG_KEY = "errorSamplingPeriod";
   private static final Logger LOGGER = LoggerFactory.getLogger(CLPLogMessageDecoder.class);
+  private static final int DEFAULT_ERROR_SAMPLING_PERIOD = 1;
```

Comment:
IMO log-per-message can be a dangerous pattern and we should make sampling period much larger by default, or makes logs debug level and let user change it at runtime when needed (metric can be used as a signal to check)

Reply:
How about 1 in 10k. messages? Modify the PR accordingly.